lib/rbi/type.rb

```diff
@@ -673,9 +895,6 @@ def proc
         Proc.new
       end
 
-      # We mark the constructor as `protected` because we want to force the use of factories on `Type` to create types
-      protected :new
-
```

Comment:
Don't we still want this especially since `Type.any(Foo, Bar)` isn't necessarily the same as `Type::Any.new(Foo, Bar)` due to the `simplify` call in `any`, for instance?